File: lib/models/room.dart
class Room {
  final String id;
  final String roomNumber;
  final int floor;
  final String status;
  final String? specialtyId;

  const Room({
    required this.id,
    required this.roomNumber,
    required this.floor,
    required this.status,
    this.specialtyId,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['_id']?.toString() ?? '',
      roomNumber: json['roomNumber']?.toString() ?? '',
      floor: json['floor'] as int? ?? 0,
      status: json['status']?.toString() ?? 'Available',
      specialtyId: json['specialtyId'] is Map 
          ? json['specialtyId']['_id']?.toString() 
          : json['specialtyId']?.toString(),
    );
  }

  Map<String, dynamic> toJson() => {
        '_id': id,
        'roomNumber': roomNumber,
        'floor': floor,
        'status': status,
        if (specialtyId != null) 'specialtyId': specialtyId,
      };
}

File: backend/models/Room.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IRoom extends Document {
  roomNumber: string;
  floor: number;
  status: 'Available' | 'Maintenance';
  specialtyId?: mongoose.Types.ObjectId;
}

const roomSchema = new Schema<IRoom>(
  {
    roomNumber: { type: String, required: true, unique: true },
    floor: { type: Number, required: true },
    status: {
      type: String,
      enum: ['Available', 'Maintenance'],
      default: 'Available',
    },
    specialtyId: { type: mongoose.Schema.Types.ObjectId, ref: 'Specialty', required: false },
  },
  { timestamps: true }
);

const Room = mongoose.model<IRoom>('Room', roomSchema);
export default Room;
